framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -277,9 +279,20 @@ protected TreeAnnotator createTreeAnnotator() {
      * @return a type annotator
      */
     protected TypeAnnotator createTypeAnnotator() {
-        implicitsTypeAnnotator = new ImplicitsTypeAnnotator(this);
+        List<TypeAnnotator> typeAnnotators = new ArrayList<>();
+        RelevantJavaTypes relevantJavaTypes =
+                checker.getClass().getAnnotation(RelevantJavaTypes.class);
+        if (relevantJavaTypes != null) {
+            Class<?>[] classes = relevantJavaTypes.value();
+            typeAnnotators.add(
+                    new IrrelevantTypeAnnotator(
+                            this, getQualifierHierarchy().getTopAnnotations(), classes));
+        }
 
-        return new ListTypeAnnotator(new PropagationTypeAnnotator(this), implicitsTypeAnnotator);
+        implicitsTypeAnnotator = new ImplicitsTypeAnnotator(this);
+        typeAnnotators.add(implicitsTypeAnnotator);
+        typeAnnotators.add(new PropagationTypeAnnotator(this));
+        return new ListTypeAnnotator(typeAnnotators.toArray(new TypeAnnotator[0]));
```

Comment:
Here we convert a list to an array and then the ListTreeAnnotator goes from the array back to a list.
Would it be easier to have a second constructor that takes a List? The varargs constructor can then create the list and then call this new constructor.